### PR TITLE
Throw if the given source file does not exist

### LINKED ISSUE #50: Does not throw on non-existing file
This test fails.
```js
test('throws on non-existing file', async t => {
    await t.throws(fn('no-file', t.context.tmp));
});
```

### PR DESCRIPTION
**Pull request by whitecrownclown**
@sindresorhus 
Fixes #50

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -51,12 +51,7 @@ module.exports = (source, destination, options = {}) => {
 		}
 
 		if (files.length === 0) {
-			progressEmitter.emit('progress', {
-				totalFiles: 0,
-				percent: 1,
-				completedFiles: 0,
-				completedSize: 0
-			});
+			throw new CpyError(`Cannot copy \`${source}\`: no files found`);
```
Comment (sindresorhus) [inline index.js]: It's only file, not multiple.

**Comment by sindresorhus** [comment]
The behavior needs to be documented in the readme and in https://github.com/sindresorhus/cpy/blob/master/index.d.ts

File: index.js [hunk 2]
```diff
@@ -51,12 +51,7 @@ module.exports = (source, destination, options = {}) => {
 		}
 
 		if (files.length === 0) {
-			progressEmitter.emit('progress', {
-				totalFiles: 0,
-				percent: 1,
-				completedFiles: 0,
-				completedSize: 0
-			});
+			throw new CpyError(`Cannot copy \`${source}\`: no file found`);
```
Comment (sindresorhus) [inline index.js]: This doesn't sound good either.
```suggestion
			throw new CpyError(`Cannot copy \`${source}\`: the file doesn't exist`);
```

Same with the readme sentence.

File: readme.md [hunk 3]
```diff
@@ -37,6 +37,8 @@ const cpy = require('cpy');
 
 Returns a `Promise<string[]>` with the destination file paths.
 
+If the file does not exist, an error will be thrown (does not apply to globs).
+
```
Comment (sindresorhus) [inline readme.md]: You need to update https://github.com/sindresorhus/cpy/blob/master/index.d.ts too.

File: test.js [hunk 4]
```diff
@@ -135,21 +135,11 @@ test('glob errors are CpyErrors', async t => {
...
-		.on('progress', event => {
-			report = event;
-		});
-
-	t.not(report, undefined);
-	t.is(report.totalFiles, 0);
-	t.is(report.completedFiles, 0);
-	t.is(report.completedSize, 0);
-	t.is(report.percent, 1);
+	const error = await t.throwsAsync(cpy(['no-file'], t.context.tmp));
```
Comment (sindresorhus) [inline test.js]: Use the object argument of `throwsAsync` instead.

File: test.js [hunk 5]
```diff
@@ -135,21 +135,12 @@ test('glob errors are CpyErrors', async t => {
...
-		.on('progress', event => {
-			report = event;
-		});
-
-	t.not(report, undefined);
-	t.is(report.totalFiles, 0);
-	t.is(report.completedFiles, 0);
-	t.is(report.completedSize, 0);
-	t.is(report.percent, 1);
+	await t.throwsAsync(cpy(['no-file'], t.context.tmp), {
```
Comment (sindresorhus) [inline test.js]: Can you add another test with 2 non-existent files as input?

Comment (whitecrownclown) [reply test.js]: Sure, I should've added it from the start.

**Comment by sindresorhus** [comment]
Can you add a test where a glob pattern doesn't match any files? I think the expected behavior there is that it should not throw.

**Comment by whitecrownclown** [comment]
@sindresorhus 	I've added the test and the previous implementation did throw so I added `hasGlob` to exclude glob patterns.

File: test.js [hunk 6]
```diff
@@ -151,6 +151,12 @@ test('throws on multiple non-existing files', async t => {
 	});
 });
 
+test('does not throw when not matching any file on glob pattern', async t => {
+	fs.mkdirSync(t.context.tmp);
+
+	await t.notThrowsAsync(cpy(['*.js'], t.context.tmp));
```
Comment (sindresorhus) [inline test.js]: Can you also add a test that tests mixed path and glob. I don't think you currently handle that correctly.

Comment (whitecrownclown) [reply test.js]: Never throw on `glob` patterns (even if mixed with `path`).

**or**

Throw on mixed `path` and `glob` when `path` does not exist?

Comment (sindresorhus) [reply test.js]: > Throw on mixed path and glob when path does not exist?

This

**Pull request merged**